src/masternodes/mn_checks.cpp

```diff
@@ -310,6 +310,17 @@ Res ApplyCreateTokenTx(CCustomCSView & mnview, CCoinsViewCache const & coins, CT
         }
     }
 
+    if (token.decimal > 18) {
+        return Res::Err("%s : %s", base, "Decimal is too deep");
+    }
+
+    CAmount coin = std::pow(10, token.decimal);
+    if (token.limit == 0) {
+        token.limit = 1200000000 * coin; // default token limit
```

Comment:
Should we use `MAX_MONEY` instead of `1200000000`?

Reply:
It's correct `MAX_MONEY / COIN * coin` but it can overflow int64 it'll be handled better in next push.